src/main/java/mate/academy/dao/BookDaoImpl.java

```
@@ -0,0 +1,121 @@
+
+            List<Book> books = new ArrayList<>();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                String title = resultSet.getObject("title", String.class);
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+
+                books.add(new Book(id, title, price));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all books. ", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sqlQuery = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery)) {
+            preparedStatement.setLong(1, id);
+
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            if (resultSet.next()) {
+                String title = resultSet.getObject("title", String.class);
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+
+                Book book = new Book(id, title, price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id = " + id + ".", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sqlQuery = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+
+            preparedStatement.executeUpdate();
```

In the `update` method, it would be beneficial to check the result of `executeUpdate()` to ensure that the update was successful. If no rows were affected, you might want to throw an exception or handle this case appropriately.